### DISCUSSION TITLE
Action keyword AND include in global search

### DISCUSSION THREAD

**Discussion by nv2p** [Flow-Launcher/Flow.Launcher#2007] [Ideas]
Hello everyone and thanks a lot to all contributors of this amazing piece of work!

I think it would be great to use action keywords not only to start a plugin, but also to filter results, especially afterward (see also this [comment](https://github.com/Flow-Launcher/Flow.Launcher/issues/769#issuecomment-991954182), which you seemed to be open to, but is apparently staled). Generally, I like to have several plugins in the main search (explorer, programs, etc.), as it's faster, because Flow Launcher remember my choices. So I can start my usual programs and search files, without the need of typing an action keyword for any of those plugins. Nevertheless, sometimes I know that I want to select a program and if the name is used in many files, I will have too many results (which I often find out afterward...). At the moment, I don't see any way to do this (correct me if I'm wrong of course and then this discussion will turn into making the feature more visible in the UI :wink:). If I define an action keyword for programs, I have to type it to see them. In PowerToys Run (which is also inspired by Wox), they implemented this by allowing you to mark a box to include in the global search:

![image](https://user-images.githubusercontent.com/49092365/227885462-27864584-0409-4950-a06c-3cd6c2361954.png)

I would be fine with this, but another way could be to allow to assign a shortcut, which apply the filter of the plugin. It would have the advantage to be a little bit faster, when applied _afterward_ (than pressing the home key, the action keyword and space), but it's maybe a little bit less practical, when you want to filter _from start_ (than just pressing the action keyword and space). So a combinaision of both? :blush:

**Comment by JoyHak** [comment]
temporary solution: I can write an autohotkey script that will replace the `:hotstring` with the program name with `apps_plugin_keyword+app_name`.
For example, enter `Explorer` to see every app related to the _explorer_. Enter `Anytype` to see just apps named _Anytype_. No keyword, just enter app name and u'll see an app. The keyword will remain in the backend...

However, I do not know if this is technically possible. It is logical that we do not want to manually add to the script the names of all programs to which we need to automatically add a `keyword`...

I need to scan a directory with links to all apps and create an array of app names in order to automatically add a keyword **only to app names**, not files... and I'm not sure that AHK will allow me to dynamically create `:hotstrings` from array elements.
